Feature/2259 replace cloudeventconfig with notif pkg utils

Floris272 opened the pull request:
Closes #2259 

**Changes**
- removes cloudeventconfig
- removes cloudevent task and uses one in notifications-api-common

This pr is waiting on #2121 and new NAC release

**Checklist**

Check off the items that are completed or not relevant.

- Experimental features/changes

  - [x] Any experimental features added in this PR are backwards compatible
  - [x] Any experimental features added in this PR are documented in the `docs/api/experimental.rst` page

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

stevenbal commented:
@Floris272 #2121 was merged, could you rebase this PR on main?

Floris272 committed:
:sparkles: [#2259] use notif api common send_cloudevent

Floris272 committed:
:sparkles: [#2259] add data attrs to base zaak cloudevents

Floris272 committed:
:bug: [#2259] fix cloudevent data urls for external objects

stevenbal commented:
Tested locally and looks good, just a question about mocks in the tests

Also realized just now that Open Notificaties doesn't support filters on `data` for cloud events yet, but I'll make a separate issue for that.

stevenbal committed:
:recycle: [#2259] Only construct cloud event data if cloud events are enabled

to avoid unnecessary queries and/or API calls to fetch catalogi, zaaktypen, etc.

stevenbal commented on src/openzaak/components/zaken/tests/test_statussen.py:
```diff
@@ -203,6 +204,7 @@ def test_create_external_statustype(self):
             mock_ztc_oas_get(m)
             m.get(statustype, json=get_statustype_response(statustype, zaaktype))
             m.get(zaaktype, json=get_zaaktype_response(catalogus, zaaktype))
+            m.get(catalogus, json=get_catalogus_response(catalogus, zaaktype))
```
Noticed that the data for the cloud events was being constructed, regardless of the value of `settings.ENABLE_CLOUD_EVENTS`, fixed it to avoid doing unnecessary queries
